polars/polars-core/src/frame/mod.rs

```diff
@@ -1073,21 +1073,26 @@ impl DataFrame {
         Ok(DataFrame::new_no_checks(new_cols))
     }
 
+    /// Drop columns that are in `names`.
     pub fn drop_many<S: AsRef<str>>(&self, names: &[S]) -> Self {
         let names: PlHashSet<_> = names.iter().map(|s| s.as_ref()).collect();
-        fn inner(df: &DataFrame, names: PlHashSet<&str>) -> DataFrame {
-            let mut new_cols = Vec::with_capacity(df.columns.len() - names.len());
-            df.columns.iter().for_each(|s| {
-                if !names.contains(&s.name()) {
-                    new_cols.push(s.clone())
-                }
-            });
+        self.drop_many_amortized(&names)
+    }
 
-            DataFrame::new_no_checks(new_cols)
-        }
-        inner(self, names)
+    /// Drop columns that are in `names` without allocating a `HashSet`.
+    pub fn drop_many_amortized(&self, names: &PlHashSet<&str>) -> DataFrame {
+        let mut new_cols = Vec::with_capacity(self.columns.len() - names.len());
```

Comment:
Can we have `names.len() > self.columns.len()` here?

Reply:
Yes, we can good catch. Can you make a PR that does a saturating sub?